### ISSUE TITLE
RFC: extract and rewrite booking page in a faster framework (svelte)

### ISSUE THREAD

**Issue by PeerRich** [calcom/cal.com#2301]
related to #2221

## Known Issues
- booking page loads the entire nextjs app
- performance can be improved
- making changes to the booking page means you'll have merge conflicts in the entire app

## What we want
- super lightweight page
- fast and great lighthouse score
- very customisable

## Proposal
- rewrite the booking page in something faster like svelte or even vanilla js
- focus on customizability for the `<script>` implementation @hariombalhara 

## Potential issues
- how does turbo repo work with different frameworks?
- do we want something non-react? is svelte worth the extra performance boost?
- will we deprecate the existing booking page or will this be additive?  (likely replace)
- what if third-party apps want to make changes to the booking page? (that could become a mess)
- an external package likely needs the public API to be ready, otherwise, we likely have CORS

**Comment by PeerRich** [comment]
alternatively, we make the booking page super duper fast with nextjs

**Comment by denik1981** [comment]
This is very related to #2237 and the way we are using turbo which is in many cases as a symlink server and not as a monorepo package manager. We must check the way we import artifacts because we might be falling in double imports which increases our final bundle size and slow down performance. Webpack does not work pretty well in this way.

As an example, if we want to work with the availability module within the `@calcom/lib` package we should avoid doing things like `import {getAvailability} from '@calcom/lib/availability'` and instead group all of our imports in a single index.ts that will export everything from the child package. There is no point using turbo as an alias manager because the mandatory question arises of `Why turbo?` if you can already organize files and folders, and add aliases to them without it.

**Comment by zomars** [comment]
Turbo serves here in more of terms of reusability. The general rule is that if the package is going to be used in multiple projects, it should be extracted to a reusable package. But agreed. We still have some code organizing to do. @denik1981

**Comment by denik1981** [comment]
Yes, agree. The tagline here is to avoid buying into a costly solution for a problem we might not have.

My humble suggestion is to treat packages as isolated packages within turbo. Make them have all the dependencies they need (not the hoisted ones) and group all the exports into a single index.ts aimed to export everything. Then benchmark once again the booking page to see if this has contributed to solving the problem. My concern is that we are maybe shipping duplicated artifacts. This was the case with Prisma in #2237 which was shipped twice and made the bundle size almost 2X.

**Comment by emrysal** [comment]
I've got pretty strong views here that we shouldn't shift framework over performance concerns. There is a LOT that can be done before we do so that can improve our booking pages, but measuring is knowing. Like @denik1981 said, our performance could be due to a variety of things (Nor do we currently use any next12 features on our booking page). 

But ultimately; with NextJS we have acquired the knowledge to have another go at our booking page and know how to spot issues with performance, with another framework like Svelte we'd need to start all over again.
